### DISCUSSION TITLE
Trouble with type-graphql migrating from apollo server 3 to 4

### DISCUSSION THREAD

**Discussion by Oskarikor98** [MichalLytek/type-graphql#1419] [Q&A]
Hello,

I'm trying to migrate from apollo server 3 to 4 before the end of life in October 2023. As of right now I'm facing a couple of issues with the process and the issues seem to be related to type-graphql. Any guidance is greatly appreciated.

Case 1:

I upgrade the apollo server to the latest (4.3.1) and both type-graphql remain at 1.1.1 & graphql remain at 15.3.0. What happens is the mutations I have will work as intended unless there is an error. So lets say I have login mutation that won't find a user -> it will crash. I believe this is due to how the error handling works in apollo server 4 as mentioned in #1397 

Case 2:

I upgrade the apollo server to the latest as well as type-graphql to 2.0.0-beta.1 and graphql to 16.6.0. Now the my service refuses to start giving me the same error as mentioned here: https://github.com/MichalLytek/type-graphql/discussions/1393#discussioncomment-4351067

In my case it should not be a typo as the path remains unchanged. Here's how I call buildSchema:

`  const apolloServer = new ApolloServer<MyContext>({
    schema: await buildSchema({
      resolvers: [__dirname + "/migration/*.ts"],
    }),
  });`

I have been reading through quite a few issues and discussions as well as apollo server documentation and am still clueless. Any help would be greatly appreciated.

**Comment by MichalLytek** [comment]
> In my case it should not be a typo as the path remains unchanged. Here's how I call buildSchema:

Glob path for resolvers will be removed in v2.0

**Reply by Oskarikor98** [reply]
Ah okay, thank you! What would be the best practice to apply the resolvers to buildSchema going 2.0 and forward?

**Reply by MichalLytek** [reply]
`resolvers: [A, B, C]` is the only option. You can construct your own glob-path import discovery util function, as long it returns resolver classes.
